MapleServer2/Packets/Helpers/ItemPacketHelper.cs

```diff
@@ -135,17 +135,22 @@ private static PacketWriter WriteAppearance(this PacketWriter pWriter, Item item
         private static PacketWriter WriteStats(this PacketWriter pWriter, ItemStats stats)
         {
             pWriter.WriteByte(); // Not part of appearance sub!
-            List<ItemStat> basicStats = stats.BasicStats.Where(x => x.GetType() == typeof(NormalStat)).ToList();
-            pWriter.WriteShort((short) basicStats.Count);
-            foreach (NormalStat stat in basicStats)
+            List<ItemStat> basicNormalStats = stats.BasicStats.Where(x => x.GetType() == typeof(NormalStat)).ToList();
+            pWriter.WriteShort((short) basicNormalStats.Count);
+            foreach (NormalStat stat in basicNormalStats)
             {
-                pWriter.Write(stat);
+                pWriter.WriteShort((short) stat.ItemAttribute);
+                pWriter.WriteInt(stat.Flat);
+                pWriter.WriteFloat(stat.Percent);
             }
-            List<ItemStat> specialBasicStats = stats.BasicStats.Where(x => x.GetType() == typeof(SpecialStat)).ToList();
-            pWriter.WriteShort((short) specialBasicStats.Count);
-            foreach (SpecialStat stat in specialBasicStats)
+
+            List<ItemStat> basicSpecialStats = stats.BasicStats.Where(x => x.GetType() == typeof(SpecialStat)).ToList();
+            pWriter.WriteShort((short) basicSpecialStats.Count);
+            foreach (SpecialStat stat in basicSpecialStats)
             {
-                pWriter.Write(stat);
+                pWriter.WriteShort((short) stat.ItemAttribute);
+                pWriter.WriteFloat(stat.Percent);
+                pWriter.WriteInt(stat.Flat);
```

Comment:
SpecialStat was wrong all this time? It isn't two floats?

Reply:
I just realized that it is two floats... but looking in game, it seems to accept it and display the value correctly for flats.

Reply:
I suppose it's the correct thing to do to change it to floats.
sadge